features/extension/src/main/java/org/apache/karaf/features/extension/StoredWiringResolver.java

```diff
@@ -111,7 +134,7 @@ synchronized void update(Bundle bundle) {
 
     synchronized void delete(Bundle bundle) {
         if (wiring.get(bundle.getBundleId()) != null) {
-            wiring.get(bundle.getBundleId()).delete(path);
+            wiring.remove(bundle.getBundleId()).delete(path);
```

Comment:
This change is unrelated with the bug. I noticed it when writing the unit tests. I wasn't expecting the deleted bundle to remain in the in memory wiring map. If it was intentional I can remove it from the commit.